p2p/speedmeter_test.go

```diff
@@ -10,10 +10,48 @@ import (
     "time"
 )
 
-func Test_speedMeter(t *testing.T) {
-    sp := NewSpeedMeter(100, 10)
-    sp.Feed(128 * 1024)
-    time.Sleep(300)
-    sp.Feed(128 * 1024)
-    fmt.Println("speedMeter:", sp.GetRate())
+func Test_NewSpeedMeter(t *testing.T) {
+    var step uint64 = 5
+    var itemsNum uint = 10
+    v1 := NewSpeedMeter(step, itemsNum)
+
+    if v1.step != step || v1.itemsNum != itemsNum {
+        fmt.Println("NewSpeedMeter Init", v1.step, v1.itemsNum)
+        t.Fail()
+    }
+}
+
+func Test_SpeedMeterFeedAndGetRate(t *testing.T) {
+    var step uint64 = 5
+    var itemsNum uint = 10
+    v1 := NewSpeedMeter(step, itemsNum)
+
+    for _, ele := range v1.itemArr {
+        if ele.tick != 0 || ele.amount != 0 {
+            fmt.Println("NewSpeedMeter ele:", ele)
+            t.Fail()
+        }
+    }
+
+    rate := v1.GetRate()
+    if rate != 0 {
+        fmt.Println("NewSpeedMeter rate:", rate)
+        t.Fail()
+    }
+
+    var num uint = 10
+    time.Sleep(1000 * time.Millisecond)
```

Comment:
why you need sleep?

Reply:
the following statement is Feed(), in which the time.Now will be called, so I use this simple way to simulate it.